addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -45,24 +62,26 @@ def _prepare_component_quantity(self, move, qty_producing):
 
     def _workorder_line_ids(self):
         self.ensure_one()
-        return self.raw_workorder_line_ids | self.finished_workorder_line_ids
+        return self.raw_workorder_line_ids | self.finished_workorder_line_ids | self.produced_workorder_line_ids
```

Comment:
can you explain why you need to make this change?

Reply:
you should fill create as many `self.finished_workorder_line_ids` as units in serial_count, then assign on all these lines a 'lot_name' (you need to add this field) as it is done in stock and then _update_finished_move should work exacty as now (needs to be adapted since i suppose it only work with a single finished line at the moment?)

Reply:
I thought it's for byproduct. I'll use finished_workorder_line_ids directly.

Reply:
finished_workorder_line_ids contained finished product and byproducts, to differentiate them we look at the product_id set on them

now with your changes it'll contain finished productS and byproducts